nixos/tests/vm-variant.nix

```
@@ -6,12 +6,13 @@ let
   evalConfig = import ../lib/eval-config.nix;
 
   nixos = evalConfig {
+    system = pkgs.stdenv.hostPlatform.system;
     modules = [
       {
         system.stateVersion = "25.05";
         fileSystems."/".device = "/dev/null";
         boot.loader.grub.device = "nodev";
-        nixpkgs.hostPlatform = pkgs.system;
+        nixpkgs.hostPlatform = pkgs.stdenv.hostPlatform.system;
```

`system` is an alias of `stdenv.hostPlatform.system` so this isn't really necessary.

---

It is if you eval with `config.allowAliase = false;`

---

Not sure, but this makes it at least possible to get the eval `name`, `version`, `pname`, `outputs`, `system`, `outputName` and `meta`.

---

@wolfgangwalther would the following be an improvement, in general? It preserves more platform configuration, but iirc you mentioned some issue with passing an already elaborated system.

```suggestion
        nixpkgs.hostPlatform = pkgs.stdenv.hostPlatform;
```

---

Got it, thank you. Marked my suggestion with a little :-1: in agreement.
I feel like we should probably fix that problem at some point. A suggestion like mine is rather tempting in any case.

The `vm-variant.nix` module is about `nixos-rebuild build-vm`. This line only affects the file it's in, not the whole test suite, so we should be good.